community-code/How_to_Monitor_Diagnose_and_Solve_Gradient_Issues_in_Foundation_Models/gradient_norm_tracking_in_pytorch .ipynb

```diff
@@ -0,0 +1,274 @@
+    "# Step 1. Load and Preprocess Dataset\n",
+    "We will use the GLUE dataset and transformers from HuggingFace. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = load_dataset(configs[\"dataset_name\"], name=configs[\"dataset_config_name\"])\n",
+    "tokenizer = BertTokenizer.from_pretrained(configs[\"model_name\"])\n",
+    "\n",
+    "\n",
+    "\n",
+    "def tokenize_function(examples):\n",
+    "    return tokenizer(\n",
+    "        examples[\"sentence1\"],\n",
+    "        examples[\"sentence2\"],\n",
+    "        truncation=True,\n",
+    "        padding=\"longest\",\n",
+    "        return_tensors=\"pt\",\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "tokenized_datasets = dataset.map(tokenize_function, batched=True)\n",
+    "tokenized_datasets.set_format(\n",
+    "    type=\"torch\", columns=[\"input_ids\", \"attention_mask\", \"token_type_ids\", \"label\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4",
+   "metadata": {},
+   "source": [
+    "# Step 2. Prepare the Dataloaders and load the Bert model for sequence classification\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_dataset = tokenized_datasets[\"train\"].shuffle(seed=configs[\"seed\"]).select(range(1000)) # Sample for demonstration\n",
+    "\n",
+    "data_collator = DataCollatorWithPadding(tokenizer=tokenizer)\n",
+    "\n",
+    "train_dataloader = DataLoader(train_dataset, shuffle=True, batch_size=configs[\"batch_size\"], collate_fn=data_collator)\n",
+    "\n",
+    "\n",
+    "from transformers import BertConfig, BertForSequenceClassification\n",
+    "\n",
+    "# Random BERT config with no LayerNorm or Dropout\n",
+    "config = BertConfig(\n",
+    "    hidden_dropout_prob=0.0,\n",
+    "    attention_probs_dropout_prob=0.0,\n",
+    "    num_hidden_layers=12,\n",
+    "    num_attention_heads=12,\n",
+    "    hidden_size=768,\n",
+    "    intermediate_size=3072,\n",
+    "    layer_norm_eps=1e-12,\n",
+    "    output_attentions=False,\n",
+    "    output_hidden_states=False,\n",
+    "    num_labels=2,\n",
+    ")\n",
+    "\n",
+    "model = BertForSequenceClassification(config)  # Not pretrained\n",
+    "model.to(device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6",
+   "metadata": {},
+   "source": [
+    "# Step 3. Initialize Neptune for Logging\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = Run(\n",
+    "    api_token=configs[\"api_token\"],\n",
+    "    project=configs[\"project\"],\n",
+    "    experiment_name=f\"gradient_tracking_lr={configs['learning_rate']}\",\n",
+    ")\n",
+    "\n",
+    "# Log configs to Neptune\n",
+    "run[\"parameters\"] = configs\n",
```

Comment:
The method to log such configs is [`log_configs()`](https://docs.neptune.ai/log_configs/).

In this case, we will need to prefix `parameters/` to the config keys so that they are all logged in the `parameters` namespace